pkg/approver/aws/aws.go

```diff
@@ -0,0 +1,168 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type awsCloud struct {
+	ec2 *ec2.EC2
+	asg *autoscaling.AutoScaling
+}
+
+func newAWSCloud(regionName string) (*awsCloud, error) {
+	creds := credentials.NewChainCredentials([]credentials.Provider{
```

Comment:
needs to work from outside the cluster too.

Reply:
This chain uses this priority ENV variable  > SharedConfigFIle (~/.aws/..) > EC2Role. This is should be valid if it needs to run outside from cluster, in which case region name would have to be provided...

Reply:
Ah, so it's all implicit and will "just work"